barretenberg/cpp/src/barretenberg/relations/translator_vm/translator_gen_perm_sort_relation.cpp

```diff
@@ -0,0 +1,132 @@
+#include "barretenberg/relations/translator_vm/translator_gen_perm_sort_relation.hpp"
+#include "barretenberg/flavor/goblin_translator.hpp"
+
+namespace proof_system {
+
+/**
+ * @brief Expression for the generalized permutation sort relation
+ *
+ * @details The relation enforces 2 constraints on each of the ordered_range_constraints wires:
+ * 1) 2 sequential values are non-descending and have a difference of at most 3, except for the value at last index
+ * 2) The value at last index is  2¹⁴ - 1
+ *
+ * @param evals transformed to `evals + C(in(X)...)*scaling_factor`
+ * @param in an std::array containing the fully extended Univariate edges.
+ * @param parameters contains beta, gamma, and public_input_delta, ....
+ * @param scaling_factor optional term to scale the evaluation before adding to evals.
+ */
+template <typename FF>
+template <typename ContainerOverSubrelations, typename AllEntities, typename Parameters>
+void GoblinTranslatorGenPermSortRelationImpl<FF>::accumulate(ContainerOverSubrelations& accumulators,
+                                                             const AllEntities& in,
+                                                             const Parameters&,
+                                                             const FF& scaling_factor)
+{
+    static const FF minus_one = FF(-1);
+    static const FF minus_two = FF(-2);
+    static const FF minus_three = FF(-3);
+    static const size_t micro_limb_bits = 14;
+    static const auto maximum_sort_value = -FF((1 << micro_limb_bits) - 1);
+
+    [&]() {
```

Comment:
As discussed, noting that since this is fighting clang format I'd prefer to make this a named function, but it's not a big deal